### ISSUE TITLE
[PRE REVIEW]: Emacs-reveal: A software bundle to create OER presentations with audio explanations

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1386]
**Submitting author:** @lechten (<a href="http://orcid.org/0000-0002-3064-147X">Jens Lechtenbörger</a>)
**Repository:** <a href="https://gitlab.com/oer/emacs-reveal" target ="_blank">https://gitlab.com/oer/emacs-reveal</a>
**Version:** v3.0.1
**Editor:** Pending
**Reviewer:** Pending

**Author instructions**

Thanks for submitting your paper to JOSS @lechten. **Currently, there isn't an JOSS editor assigned** to your paper.

@lechten if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

For a list of things I can do to help you, just type:

```
@whedon commands
```

**What happens now?**

This submission is currently in a `pre-review` state which means we are waiting for an editor to be assigned and for them to find some reviewers for your submission. This may take anything between a few hours to a couple of weeks. Thanks for your patience :smile_cat:

You can help the editor by looking at [this list of potential reviewers](https://bit.ly/joss-reviewers) to identify individuals who might be able to review your submission (please start at the bottom of the list). Also, feel free to suggest individuals who are not on this list by mentioning their GitHub handles here.

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1386 with the following error: 

 /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `block in find': No such file or directory - tmp/1386 (Errno::ENOENT)
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `collect!'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `find'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-a1723d160bb6/lib/whedon/processor.rb:57:in `find_paper_paths'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-a1723d160bb6/bin/whedon:50:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-a1723d160bb6/bin/whedon:116:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by lechten** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

```

**Comment by lechten** [comment]
Concerning potential reviewers: I did not see anyone advertising Open Educational Resources. Here are some suggestions:

- @adammichaelwood You mention Continuous Deployment, Markup Languages, and Education, all of which are relevant in this paper.
- @bradweiner and @jrosen48 I hope that you find this paper interesting based on your educational topics.
- @markgalassi Maybe your interest in Lisp (the language of emacs-reveal) and Software Engineering(CI/CD deployment on GitLab) makes the paper interesting for you?

**Comment by jrosen48** [comment]
Hi, I'd be happy to serve as a reviewer for this software bundle & paper.

**Comment by markgalassi** [comment]
I'm also happy to serve as a reviewer for emacs-reveal

**Comment by bradweiner** [comment]
Hi all,

Thanks for considering me as a reviewer. At this point, I don’t have the capacity to do this review. My apologies, but I have a few other commitments that are taking my time and I wouldn’t want it to get held up by me.

It looks like a very cool project.

Brad


Brad Weiner, Ph.D.
Senior Director of Data Science, Capture Higher Ed

p:      612.208.8852      |     e:      bweiner@capturehighered.com<mailto:bweiner@capturehighered.com>
w:      www.capturehighered.com<http://www.capturehighered.com>
________________________________

<https://www.facebook.com/CaptureHigherEd>
Facebook
<https://twitter.com/CaptureHigherEd>
Twitter
<https://www.linkedin.com/company/capture-higher-ed>
LinkedIn






On Apr 15, 2019, at 6:38 AM, Jens Lechtenbörger <notifications@github.com<mailto:notifications@github.com>> wrote:


Concerning potential reviewers: I did not see anyone advertising Open Educational Resources. Here are some suggestions:

  *   @adammichaelwood<https://github.com/adammichaelwood> You mention Continuous Deployment, Markup Languages, and Education, all of which are relevant in this paper.
  *   @bradweiner<https://github.com/bradweiner> and @jrosen48<https://github.com/jrosen48> I hope that you find this paper interesting based on your educational topics.
  *   @markgalassi<https://github.com/markgalassi> Maybe your interest in Lisp (the language of emacs-reveal) and Software Engineering(CI/CD deployment on GitLab) makes the paper interesting for you?

**Comment by lechten** [comment]
Hi all, many thanks for your fast responses and many thanks to @jrosen48 and @markgalassi for being willing to work on my behalf!
Is there anything for me to do now? What about the PDF compilation failure?

**Comment by arfon** [comment]
:wave: @lechten thanks for your submission to JOSS. From a quick inspection of this submission it's not entirely obvious that it meets our [submission criteria](https://joss.readthedocs.io/en/latest/submitting.html#submission-requirements). In particular, this item:

> - Your software should have an obvious research application

Could you confirm here that there _is_ a research application for this software (and explain what that application is)? The section [_'what should my paper contain'_](http://joss.theoj.org/about#paper_structure) has some guidance for the sort of content we're looking to be present in the `paper.md`.

Many thanks!

**Comment by lechten** [comment]
There are different directions of research.  First, the Action Plan [@Une17] shows a gap for OER in practice, also identifying a lack of tools.  How should “good” tools look like?  Prior work [HWSJ10] proposed the ALMS framework to specify desirable technical properties for OER.  In [Lec19] I extended that framework based on more general ideas of software engineering and technical writing.  I believe this to touch fundamental questions of how to produce and use OER.

Second, in the paper I mention CC REL [AAL+12].  In work currently under review, I show CC REL not to be sufficient to express license attribution in a machine-readable fashion.  In emacs-reveal, an ad-hoc extension of CC REL is implemented, which allows to gather experiences in a real-world testbed.  Further standardization is necessary, but currently nobody seems to know what is necessary.

**Comment by arfon** [comment]
OK thanks for the background. I'd like to consult with the JOSS editorial team to get their feedback too.

Before I do this however, could you clarify what and where the source code is for this submission? I can see a couple of `.el` files but there seems to be a Docker file referenced too. Could you clarify?

**Comment by lechten** [comment]
Thanks for your questions, which help me to position emacs-reveal for a wider audience regardless of how this submissions plays out.

You can think of emacs-reveal as a plugin for GNU Emacs to generate HTML presentations from input files in [Org mode](https://orgmode.org/).  Emacs plugins/libraries/packages are written in Emacs Lisp (`el` files).  If you are disappointed by the amount of code in the repository: In the past (until the [beginning of January 2019](https://gitlab.com/oer/emacs-reveal/tree/cfcdf4449df51b0137fb6ed0e3e00c9b05466b32)) it contained more code and resources.  Based on user feedback, I made the bulk of the code available as separate Emacs packages `oer-reveal` and `org-re-reveal-ref` (on GitLab as well as on the standard repository [MELPA](https://melpa.org/) for such purposes).  Nowadays, at the top of file [emacs-reveal.el](https://gitlab.com/oer/emacs-reveal/blob/master/emacs-reveal.el#L11) you see `oer-reveal` and `org-re-reveal-ref` listed as requirements, and the file just contains necessary setup code to pull various bits and pieces together.

The Docker image allows to use emacs-reveal in a controlled environment.  For example, the  [Howto of emacs-reveal](https://oer.gitlab.io/emacs-reveal-howto/howto.html) is generated by a [GitLab CI runner based on that Docker image](https://gitlab.com/oer/emacs-reveal-howto/blob/master/.gitlab-ci.yml).

**Comment by lechten** [comment]
I'd like to add one point concerning research: I assign self-study tasks to students based on presentations generated with emacs-reveal (in Just-in-Time Teaching, which shares similarities with Flipped Classroom approaches to foster active learning). Frequently, such approaches are based on video input, which I dislike personally (no skim reading, no hyperlinks, difficult navigation, difficult re-use). However, what is really better for learning? [Lec19] includes a survey result showing a slight preference by students for emacs-reveal presentations over video. This is certainly not the typical type of research for JOSS, but research enabled by FLOSS anyways.

**Comment by arfon** [comment]
Hi @lechten - many thanks for all of your feedback on my questions here. I've discussed this submission with the JOSS editorial team and we've decided that this not in scope for JOSS.

We have a sister journal - Journal of Open Source Education (http://jose.theoj.org/) that this submission might be a better fit for if that's interesting to you?

Thanks again for your interest in publishing with JOSS.

**Comment by lechten** [comment]
Thanks for the pointer.  I'll check it out.
